frontend/src/components/CalcSpectrum.tsx

```diff
@@ -163,7 +166,40 @@ export const CalcSpectrum: React.FC = () => {
           setCalcSpectrumResponse(response);
         }
       }
-      setLoading(false); //setLoading(false) is called after the response is received
+      setLoading(false);
+      setdownloadbutton(false);
+    });
+  };
+  //download the .spec file
+  const downloadSpec = async (data: FormValues): Promise<void> => {
```

Comment:
Both this method an calculate-spectrum endpoints do the exact same operations with the exact same payload but with a different endpoint. Could we abstract them both to a common function that takes as input either "calculate-spectrum" or "download_spectrum"?

Reply:
I think it is great to have separate ones. but if want I can do that.

Reply:
I'd follow @suzil advices there